File: 3_Signalanalyse/Python/gravitational_waves.py
import numpy as np
import matplotlib.pyplot as plt
from wavelet_transform import wavelet_transform

# load data
lines = np.loadtxt("H1.txt", comments="#")
t = lines[:, 0]
y = lines[:, 1]
sampl_freq = 1/(t[1]-t[0])

plt.figure()
plt.plot(t, y)
plt.xlabel('Time $t$ / s')
plt.ylabel('Strain $\epsilon\cdot 1e-21$')

delta_f = sampl_freq/len(y)
f = np.arange(-sampl_freq/2, sampl_freq/2, delta_f)
y_fft = abs(np.fft.fftshift(np.fft.fft(y)))

plt.figure()
plt.plot(f, y_fft)
plt.xlabel('Frequency $f$ / Hz')
plt.ylabel('Betragsspekrum')


# %%
# define wavelet parameters
f_c = 50
f_b = 1/50/2
scales = np.linspace(1/10, 10, num=1000)


def wavelet_fun(f):
    # define wavelet function (complex morlet)
    return np.exp(-f_b ** 2 * np.pi ** 2 * (f - f_c) ** 2)

# compute wavelet transform
wt = wavelet_transform(y, sampl_freq, scales, wavelet_fun)

plt.figure()
plt.pcolormesh(t, f_c/scales, abs(wt))
plt.xlabel('Time $t$ / s')
plt.ylabel('Frequency $f$ / Hz')

plt.show()

File: 3_Signalanalyse/MATLAB/gravitationalWaves.m
clear all
close all
clc

% load data
[t, y] = textread('H1.txt', '%f %f', 'commentstyle', 'shell');

figure
plot(t, y)
xlabel('time {\itt} / s')
ylabel('strain \epsilon \cdot 1e21')

sampl_freq = 1/(t(2)-t(1));

% scaling parameter a
scales = linspace(1/10, 10, 1000);  

% define wavelet and its paramters
fb = 1/50/2;
fc = 50;
waveletFun = @(f) exp(-fb^2 * pi^2 *... 
        (f - fc).^2);

% caluclate Wavelet Transform
wt = waveletTransform(y, sampl_freq, scales, waveletFun);

f = fc./scales;
figure
pcolor(t, f, abs(wt)), shading flat
xlabel('Time {\itt} / s')
ylabel('Frequency {\itf} / Hz')

%%
sig.val = y;
sig.period = 1/sampl_freq;
cwtstruct = cwtft(sig);